bin/desi_daily_archive.sh

```diff
@@ -0,0 +1,119 @@
+    local job_name="redux_daily_tiles_archive_${tileid}_${archivedate}"
+    cat > ${jobs}/${job_name}.sh <<EOT
+#!/bin/bash
+#SBATCH --account=desi
+#SBATCH --qos=xfer
+#SBATCH --constraint=cron
+#SBATCH --time=30:00
+#SBATCH --job-name=${job_name}
+#SBATCH --output=${jobs}/%x-%j.log
+#SBATCH --licenses=cfs,scratch
+source /global/common/software/desi/desi_environment.sh main
+module load desida desiBackup
+source \${DESIDA}/bin/desida_library.sh
+set -o xtrace
+shopt -s extglob
+if [[ "${logs}" == "True" ]]; then
+    cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive/${tileid}/${archivedate}/logs
+    sha256sum * > ${scratch}/${job_name}_logs.sha256sum
+    unlock_and_move ${scratch}/${job_name}_logs.sha256sum
+fi
+cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive/${tileid}/${archivedate}
+sha256sum !(logs) > ${scratch}/${job_name}.sha256sum
+unlock_and_move ${scratch}/${job_name}.sha256sum
+cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive
+htar -cvf desi/spectro/redux/daily/tiles/archive/${job_name}.tar -H crc:verify=all ${tileid}/${archivedate}
+[[ \$? == 0 ]] && mv -v ${jobs}/${job_name}.sh ${jobs}/done
+EOT
+    chmod +x ${jobs}/${job_name}.sh
+}
+#
+# Get options.
+#
+batch=true
+jobs=${DESI_ROOT}/users/${USER}/jobs
+scratch=${SCRATCH}
+verbose=false
+while getopts Bhj:s:Vv argname; do
+    case ${argname} in
+        B) batch=false ;;
+        h) usage; exit 0 ;;
+        j) jobs=${OPTARG} ;;
+        s) scratch=${OPTARG} ;;
+        V) version; exit 0 ;;
+        v) verbose=true ;;
+        *) usage; exit 1 ;;
+    esac
+done
+#
+# Create job directories.
+#
+[[ -d ${jobs}/done ]] || mkdir -p ${jobs}/done
+#
+# Find TILEID/ARCHIVEDATE directories.
+#
+batch_jobs_created='False'
+archive_dir=${DESI_SPECTRO_REDUX}/daily/tiles/archive
+for tile_dir in ${archive_dir}/*; do
+    tileid=$(basename ${tile_dir})
+    for archivedate_dir in ${tile_dir}/*; do
+        archivedate=$(basename ${archivedate_dir})
+        checksum_file=redux_daily_tiles_archive_${tileid}_${archivedate}.sha256sum
+        if [[ -f ${archivedate_dir}/${checksum_file} ]]; then
+            ${verbose} && echo "DEBUG: ${archivedate_dir}/${checksum_file} detected, skipping ${tileid}/${archivedate}."
+        else
+            echo "INFO: ${tileid}/${archivedate} will be backed up."
+            logs='False'
+            if [[ -d ${archivedate_dir}/logs ]]; then
+                ${verbose} && echo "DEBUG: ${tileid}/${archivedate}/logs will be checksummed."
+                logs='True'
+            fi
+            ${verbose} && echo "DEBUG: create_archivedate_job ${tileid} ${archivedate} ${logs}"
+            create_archivedate_job ${tileid} ${archivedate} ${logs}
+            batch_jobs_created='True'
+        fi
+    done
+done
+#
+# Submit a workflow job that will submit the batch jobs.
+#
+if [[ "${batch_jobs_created}" == "True" ]]; then
+    ${verbose} && echo "DEBUG: sbatch ${jobs}/submit_daily_tiles_archive.sh"
+    ${batch} && sbatch ${jobs}/submit_daily_tiles_archive.sh
```

Comment:
What is the motivation for submitting a workflow job to submit more jobs, vs. having this script directly submit the xfer jobs?

Reply:
It's a little dangerous to just submit a potentially large number of jobs to the `xfer` queue, because those jobs can block other jobs from running. The safe way is to submit jobs with a separate script that ensures that fewer than 15 jobs (per user) are running at any one time. That way some slots in the `xfer` queue are still available for unrelated jobs.